syntax = "proto3";
package posts.v1;


service Posts {
    rpc GetPost(GetPostRequest) returns (GetPostResponse) {}
    rpc GetPosts(GetPostsRequest) returns (GetPostsResponse) {}
    rpc CreatePost(CreatePostRequest) returns (CreatePostResponse) {}
    rpc UpdatePost(UpdatePostRequest) returns (UpdatePostResponse) {}
    rpc DeletePost(DeletePostRequest) returns (DeletePostResponse) {}
}

message GetPostRequest {
    string id = 1;
}

message Post {
    string id = 1;
    string title = 2;
    string content = 3;
    string created_at = 4;
    string updated_at = 5;
}

message GetPostResponse {
    Post post = 1;
}


message GetPostsRequest {}

message GetPostsResponse {
    repeated Post posts = 1;
}

message CreatePostRequest {}

message CreatePostResponse {
    string id = 1;
}

message UpdatePostRequest {
    Post post = 1;
}

message UpdatePostResponse {}

message DeletePostRequest {
    string id = 1;
}

message DeletePostResponse {}